### PR TITLE
Fixed bug where squaring values wouldn't enforce positivity

### PR DESCRIPTION
**Pull request by jrmccluskey**

### CODE REVIEW

**Comment by polewczakp** [comment]
I can't think of example where squaring a number would give negative value except comples numbers, eg. sqrt(-1). Could you provide an example, please?

**Comment by jrmccluskey** [comment]
I'll be honest this was a year ago and I don't have any of the data I was running through at the time, so I don't have an example (or any specific idea of what caused negative values.) A complex number being produced by the pipeline makes sense as to why negative values would get thrown out by squaring though.

**Comment by polewczakp** [comment]
Obviously there is probably no audio waves with complex numbers but I came to the idea that your array could have been square-rooted before processing :)

File: pyAudioAnalysis/audioFeatureExtraction.py [hunk 1]
```diff
@@ -25,12 +25,12 @@ def stZCR(frame):
 
 def stEnergy(frame):
     """Computes signal energy of frame"""
-    return numpy.sum(frame ** 2) / numpy.float64(len(frame))
+    return numpy.sum(numpy.absolute(frame ** 2)) / numpy.float64(len(frame))
```
Comment (Tronic) [inline pyAudioAnalysis/audioFeatureExtraction.py:28]: Looks like unnecessary use of numpy-namespaced functions. I believe this is equivalent to

```python
return abs(frame**2) / len(frame)
```

... or as noted, given that frame is an array of float and not complex, the `abs` can be removed as well. If performance is a concern, array multiplication runs significantly faster:

```python
return 1.0 / len(frame) * frame**2
```

**Comment by slowglow** [comment]
I was about to suggest closing this pull request, as even the original author cannot verify where or why the code failed.
Then I found a thread that may throw some light on the issue.
Might it be that there was overflow during taking a root of a signal, as [discussed here](https://stackoverflow.com/questions/17197492/is-there-a-library-function-for-root-mean-square-error-rmse-in-python).

If not, then maybe closing this PR is appropriate.

**Comment by jrmccluskey** [comment]
I thought I had closed it, sorry about that.

**Pull request closed without merging**